overlord/servicestate/quota_handlers.go

```diff
@@ -596,6 +596,16 @@ func ensureSnapServicesStateForGroup(st *state.State, grp *quota.Group, opts *en
 }
 
 func validateSnapForAddingToGroup(st *state.State, snaps []string, group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
```

Comment:
the name of this function is now slightly misleading, it validates not just that the specified snaps can be added to a group, but also that the group "can be updated", since if I try to set a new resource limit like log namespacing on an existing mixed group of snaps and sub-groups, this function will prevent that even though no "snaps" are being validated here

Reply:
I don't like multi-purpose functions, so how about I only do the check if snaps is not empty? Then the check is indeed correct and only triggers in case we are adding snaps to the group. Or how do we want to react in case we are trying to add a snap in a pre-existing group that already has both?

Reply:
We should not allow modifying in any way (except perhaps deleting) a quota group that has a mix of snaps and sub-groups. 

If you wanted, you could just split this check into a different function that is called in all the same codepaths that we have today about modifying/updating quota groups I suppose

Reply:
Ok great - i'll update it, thanks Ian